### DISCUSSION TITLE
How to add new problems

### DISCUSSION THREAD

**Discussion by Lessvn** [ds4dm/ecole#242] [Instance Generators]
Hi, 

I am new to ML/CO and my question my seem basic.  I am going to solve vehicle routing problem (VRP). How I build my model and set up the instance generator for? I am coding in Python, so should I code it in SCIP? What steps are needed?  Thanks!

**Comment by AntoinePrv** [comment]
Hello @Lessvn,

To define your problem, you can either use one of the file type supported by SCIP, or use the modelling methods of PySCIPOpt (`Model.addVar`, `Model.addCons`, `Model.setObsejctive`...).

Then you can convert it to an Ecole Model using `ecole.scip.Model.from_pysciopt`.
